tutorials/intro-to-swagger-api.md

```diff
@@ -0,0 +1,109 @@
+       2. Re-create the generated code
+
+### (API-SWAG-2) Add/Remove A Parameter From An Endpoint 
+In the previous task we generated the swagger java code and compile our handwritten code.
+Let's explore some of our handwritten code and see how it interacts with the Swagger/OpenAPI code
+1. Go to `src/main/java/org/pmiops/workbench/api/WorkspacesController.java`
+   1. Let's look at one of the overriden methods in this file - `getWorkspace`
+   2. Add a parameter to this method. Try compiling - what do you think will happen when you compile?
+   3. Did your IDE respond to the change? (If not check your IDE settings)
+2. Undo your changes above, this time remove a parameter from the method.
+
+### (API-SWAG-3) Bring up the server and inspect an endpoint
+1. Run `./project.rb dev-up`
+   This should bring up the API server allowing you to send requests to the API endpoints
+2. Using a tool like [insomnia](https://insomnia.rest/) query the endpoints.
+   You will need to generate a Bearer token to query the endpoint.
+   You can do this with the gcp CLI command: `gcloud auth print-access-token`
+   Make sure you are logged in to your pmi-ops account on the CLI. You can activate this account with `gcloud config activate psantos@fake-research-aou.org`
+3. An endpoint you can inspect is `http://localhost:8081/v1/workspaces`
+   This will return all workspaces in your account
+
+### (API-SWAG-4) Update the endpoint to accept a new parameter
+1. Open the `workbench-api.yaml` and find the endpoint whose `operationId` is `getWorkspace`
+   1. Examine the structure in the yaml file. Note the parameters.
+2. Check out the super implementation of `getWorkspace` (`api/build/swagger2/org/pmiops/workbench/api/WorkspacesApiDelegate.java`)
+   1. This is the generated code from swagger. It generates this code based on the API definitions in the yaml file, in this case it is the `workbench-api.yaml`
+3. Back in `workbench-api.yaml`...
+   1. Add a [parameter](https://swagger.io/docs/specification/describing-parameters/) to the `get` method of the API. You can take a look at the `patch` method of the `/v1/workspaces/{workspaceNamespace}/{workspaceId}` path for a live example
+   2. Make the parameter should be a string in the body of the request. You can play around more with parameters after.
+   3. Name your parameter "test"
+4. Open `api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java`
+   1. Go to the `getWorkspace method`
+   2. You will need to add the parameter to this method.
+5. Restart the server (API-SWAG-3) and inspect your endpoint
+   1. When calling the endpoint, be sure to add your parameter to the JSON body: `{"test": "Hello"}`
+   2. What happens when you omit this required parameter?
+
+### (API-SWAG-5) Add a new endpoint
+1. Open the `workbench-api.yaml` and find the beginning of the endpoint definitions.
+2. Create a path for your endpoint, for the purposes of this example I will call the endpoint `/v1/workspaces/thing`
```

Comment:
This is inviting complications / interference with `GET /v1/workspaces/{ns}/{id}`. I would just give it a more distinctive path that doesn't risk conflicting.